### DISCUSSION TITLE
i need to get the value  select field to populate a text input from  deferent table , i try my best

### DISCUSSION THREAD

**Discussion by yassin459dz** [filamentphp/filament#13812] [Help]
### Package

Form builder

### Package Version

v3.2.96

### How can we help you?

                        Select::make('client_id')
                        ->label('Client')
                        ->relationship('client', 'name')
                        ->searchable()
                        ->preload()
                        ->createOptionForm([
                            TextInput::make('name')->required(),
                            TextInput::make('phone')->required(),
                        ])
                        ->editOptionForm([
                            TextInput::make('name')->required(),
                            TextInput::make('phone')->required(),
                        ])
                        ->reactive()
                        ->afterStateUpdated(function ($state, $set) {
                            $client = Client::find($state);
                            if ($client) {
                                $firstCar = $client->cars()->first();
                                if ($firstCar) {
                                    $set('car_id', $firstCar->id);
                                } else {
                                    $set('car_id', null);
                                }

                                // Update the car options based on the selected client
                                $carsRelated = $client->cars()->pluck('model', 'id')->toArray(); // Adjust 'model' to the correct column name
                                $otherCars = Car::whereNotIn('id', array_keys($carsRelated))->pluck('model', 'id')->toArray(); // Adjust 'model' to the correct column name
                                $carOptions = [
                                    'Related Cars' => $carsRelated,
                                    'Other Cars' => $otherCars,
                                ];
                            } else {
                                $carOptions = [
                                    'Related Cars' => [],
                                    'Other Cars' => Car::pluck('model', 'id')->toArray(), // Adjust 'model' to the correct column name
                                ];
                            }
                            $set('car_options', $carOptions);
                        }),

                    Select::make('car_id')
                        ->label('Car')
                        ->searchable()
                        ->options(function ($get) {
                            return $get('car_options') ?: [];
                        })
                        ->reactive(),

                        Group::make()
                            ->relationship('matricule')
                            ->schema([
                                TextInput::make('mat')
                                    ->required(),
                                TextInput::make('km')
                                    ->required()
                                    ->dehydrated(),

                                TextInput::make('client_id')
                                    ->required(),
                                TextInput::make('car_id')
                                    ->required(),
                                    ])
                            ]),

**Comment by yassin459dz** [comment]
need help pls

**Reply by leandrocfe** [reply]
what is the issue?

**Reply by yassin459dz** [reply]
I want to retrieve data from the matricule table and display it in a text field when I select a car in a Filament form. I’m using three different tables: client, car, and matricule, and they are all related to each other. When a car is selected, I want to show the corresponding data from the matricule table in these text fields. If the data is not null, it should be displayed.
